app/src/main/java/org/mozilla/focus/topsites/TopSitesOverlay.kt

```diff
@@ -82,6 +74,23 @@ fun TopSitesOverlay(modifier: Modifier = Modifier) {
     }
 }
 
+private fun openTopSite(item: TopSite, components: Components) {
+    val currentTabId = components.store.state.selectedTabId
+    if (currentTabId.isNullOrEmpty()) {
+        components.tabsUseCases.addTab(
+            url = item.url,
+            source = SessionState.Source.Internal.HomeScreen,
+            selectTab = true,
+            private = true,
+        )
+    } else {
+        components.sessionUseCases.loadUrl(item.url, currentTabId)
+        components.appStore.dispatch(AppAction.FinishEdit(currentTabId))
```

Comment:
Interesting, I see we do this in the UrlInputFragment as well even when we are dismissing it without action.